### ISSUE TITLE
Support for PHP 8.1?

### ISSUE THREAD

**Issue by edgreenberg** [PHPOffice/PhpSpreadsheet#3854]
This is:

```
- [ ] a bug report
- [ x ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
PhpSpreadsheet should run on PHP 8.1, since that's what's available on Ubuntu 22.04 (the latest)

### What is the current behavior?
Package supports only 8.0 or 7.x

### What are the steps to reproduce?
Try to install using composer when php only 8.1 is on the machine. 

### Which versions of PhpSpreadsheet and PHP are affected?
1.29.0

**Comment by pjaudiomv** [comment]
I have been pinning my composer to dev-master to get around this but I too am eagerly awaiting the next release to be able to pin back to a known stable version that supports 8.1

**Comment by oleibman** [comment]
PhpSpreadsheet runs on Php 8.0, 8.1, 8.2, and 8.3. Our unit test suite, which has definitely run today, runs on all of these (and nightly versions of Php 8.4). What makes you think it doesn't run on 8.1?

**Comment by pjaudiomv** [comment]
it runs fine but doesn't install fine using composer package

**Comment by pjaudiomv** [comment]
the latest tagged release uses htmlpurifier version 4.16 which doesnt support 8.1 so there is a clash, when that was removed for anti-xss here https://github.com/PHPOffice/PhpSpreadsheet/commit/4e54ed389b0bb74b79a0f312a6ecc3cf35cb7de6 it resolved this issue

**Comment by oleibman** [comment]
I'm not sure why that should be. The file composer.json for PhpSpreadsheet 1.29.0 specifies in its requirements:
```
"ezyang/htmlpurifier": "^4.15",
```
So, as far as I know, if you are running Composer with Php8.1 and htmlpurifier 4.1.16 and 4.1.16 doesn't support Php8.1, it should find a later version. However, even that doesn't seem necessary - here's what's in composer.lock for PhpSpreadsheet 1.29.0:
```
"name": "ezyang/htmlpurifier",
"version": "v4.16.0",
...
            "require": {
                "php": "~5.6.0 || ~7.0.0 || ~7.1.0 || ~7.2.0 || ~7.3.0 || ~7.4.0 || ~8.0.0 || ~8.1.0 || ~8.2.0"
            },
```
So it seems that it should support Php 8.1 anyhow. I'm not a composer expert, so, if you think my analysis here is wrong, please let me know why. At any rate, our unit test suite ran with htmlpurifier for Php 8.1, 8.2, and unreleased 8.3 for quite some time without any issues.

**Comment by pjaudiomv** [comment]
your analysis is right. im sorry i apologize and i apologize for hijacking this thread. My issue was with 8.3 which didn't work in early november when i last tried but ezyang/htmlpurifier released 4.17 on nov 17th with 8.3 added. Thank you for your time and clarifying.

**Comment by PowerKiKi** [comment]
PhpSpreadsheet is compatible with PHP 8.1.
